### ISSUE TITLE
What do you think about running `miri` as part of CI?

### ISSUE THREAD

**Issue by jeffwidman** [DataDog/glommio#309]
Given that this project uses unsafe, what do you think about adding `miri` to the CI pipeline? 

A little background on `miri`: https://www.ralfj.de/blog/2020/09/28/miri.html

In fact, `miri` was already used to [find a memory leak in `glommio`](https://github.com/DataDog/glommio/issues/110).

There's more info [here on how to integrate it into a CI pipeline](https://github.com/rust-lang/miri/#running-miri-on-ci), and I'd be happy to take a stab at it, but wanted to run it by the maintainers before going to the effort of trying to wire it up.

Also, there's a chance it may be too slow to run on every PR. I don't know if `circleci` supports running a nightly job against `master`, but that may be more efficient from a development standpoint.

What do you think?

**Comment by bryandmc** [comment]
Yeah it would be awesome if it was possible.. cause I have heard goo things about miri. But how would we get around all the syscalls and then there's some other limitations as well that would need to be resolved. If they were I'd be all for it but as mentioned above I'd also just love to have the tests running in CI as well.

**Comment by jeffwidman** [comment]
Makes sense... I opened https://github.com/DataDog/glommio/issues/324 to track getting the actual tests running in CI. Whenever that gets resolved then can circle back to look into this further.
